okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/DnsOverHttps.kt

```diff
@@ -67,7 +69,31 @@ class DnsOverHttps internal constructor(
       }
     }
 
-    return lookupHttps(hostname)
+    return try {
+      lookupHttps(hostname)
+    } catch (bestFailure: UnknownHostException) {
+      if (checkIsIpAddress(hostname)) {
+        try {
+          InetAddress.getAllByName(hostname).toList()
+        } catch (e: UnknownHostException) {
+          throw bestFailure
+        }
+      } else {
+        throw bestFailure
+      }
+    }
+  }
+
+  private fun checkIsIpAddress(hostname: String): Boolean {
+    if (ipv4Regex.matches(hostname)) {
+      return true
+    }
+
+    if (hostname[0].digitToIntOrNull(16) != null || hostname[0] == ':' || hostname[0] == '[') {
```

Comment:
If the hostname is `[`, that’s malformed. We do not pass the square braces to `Dns`.

Reply:
I thought that the getAllByName will strip the square braces so I just pass the string to getAllByName directly.
Due to the canParseAsIpAddress will not handle `[`, I will strip the square braces and use the existing one to check if the hostname is a IP address.